### PR TITLE
Update theme to 0.2.x

### PR DESCRIPTION
**Pull request by yathomasi**
Similar to https://github.com/iterative/dvc.org/pull/4118, https://github.com/iterative/cml.dev/pull/402 for `mlem.ai`.

### CODE REVIEW

**Review by rogermparent — COMMENTED** [review]
Nice work! I noticed some things while going through each page type that'll need to be changed, but this is most of the way there. Any subjective quibbles I have with implementation I didn't bother to list and can be handled later.

Homepage:

- Mobile and tablet look identical! :tada: 

- Large padding added to logos on desktop screen
image vs 
image

Docs:
- Extra padding in docs sidebars
image

- Sticky sidebars broken when scrolled
image

- Command anchors could use `scroll-padding`
image

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+yarn lint-staged
```
Comment (yathomasi) [inline .husky/pre-commit:4]: Just noticed the husky pre-commit was not working. It was simple change to add to this PR

File: src/@dvcorg/gatsby-theme-iterative/components/MainLayout/base.css [hunk 2]
```diff
@@ -96,3 +104,7 @@ body.user-is-tabbing .link-with-focus {
 * {
   text-size-adjust: none;
 }
+
+[class^='styles-module--innerSidebar'] {
+  @apply pt-20;
+}
```
Comment (yathomasi) [inline src/@dvcorg/gatsby-theme-iterative/components/MainLayout/base.css:110]: This one could be the kinda hacky way but I didn't want to shadow the entire CSS file for a single change.

**Review by yathomasi — COMMENTED** [review]
Thanks, @rogermparent for the review and help with this PR. I have covered all from the review list. PTAL

**Review by rogermparent — COMMENTED** [review]
Great work! The last review's issues were all taken care of. I don't know if I missed this or it's an effect of padding removal: the top margin is a but uncomfortably close to the top on the homepage on tablet/desktop.

image

~~Still some stickiness issues on mobile docs:~~ nevermind, that's prod :sweat_smile: 
image

**Comment by rogermparent** [comment]
image
We get a weird half-mobile half-desktop state on the sidebar at ~500px

**Comment by yathomasi** [comment]
> Great work! The last review's issues were all taken care of. I don't know if I missed this or it's an effect of padding removal: the top margin is a but uncomfortably close to the top on the homepage on tablet/desktop.

Thanks for pointing these out. The top bar was updated to use the same approach used by the theme. The default height in the theme is smaller. I did add the margin but for sure we can fine-tune it more.

> We get a weird half-mobile half-desktop state on the sidebar at ~500px

Looks like I overlooked the middle screen sizes. I will make sure of them. Thanks.

**Comment by rogermparent** [comment]
Thanks for pointing these out. The top bar was updated to use the same approach used by the theme. The default height in the theme is smaller. I did add the margin but for sure we can fine-tune it more.

Ah yeah, the new version of the theme has the children of the layout provide their own padding, particularly to get the docs sidebars working because they specifically don't want any top screen padding.

**Review by rogermparent — APPROVED** [review]
Looks like all my issues from last review are solved, and I don't see any more on another look around!

**Comment by yathomasi** [comment]
> Looks like all my issues from last review are solved, and I don't see any more on another look around!

Great! Thanks again for the help and review throughout this PR.

**Pull request merged**